### PR TITLE
fix(appsync): `ttl` property of `CachingConfig` is not required

### PR DESCRIPTION
**Pull request by otaviomacedo**
According to the current CloudFormaton spec for [AWS::AppSync::Resolver CachingConfig][1], the `Ttl` property is not required. But if it's not provided, the deploy will fail with the message:

>  TTL value cannot be smaller than 1 second and larger than 3600 seconds. (Service: AWSAppSync; Status Code: 400; Error Code: BadRequestException

The AppSync team has confirmed that the property is indeed required and they will push a change to the spec. This change is proactively making the property required on the CDK.

Fixes https://github.com/aws/aws-cdk/issues/17925.

BREAKING CHANGE: The `CachingConfig#ttl` property is now required. 

[1]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appsync-resolver-cachingconfig.html#cfn-appsync-resolver-cachingconfig-ttl
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by njlynch — COMMENTED** [review]
Two thoughts:

1. Should we wait to make this required in the CDK until the spec change has been merged? Maybe not, but coordinating the two would make some sense.
2. Either way, this needs to be marked as a BREAKING CHANGE.

**Comment by otaviomacedo** [comment]
> 1. Should we wait to make this required in the CDK until the spec change has been merged? Maybe not, but coordinating the two would make some sense.

I don't mind. We can hold this off until the new spec is out. On the other hand, if this property is left undefined, it will fail one way or the other. With this change, it will fail faster (compile time).

> 2. Either way, this needs to be marked as a BREAKING CHANGE.

Done.

**Review by njlynch — APPROVED** [review]

**Comment by otaviomacedo** [comment]
Latest update from the AppSync team:

> The recommended workaround at this point is to include the TTL field. Our next step is to make this field required and update our docs.

Removing the `pr/do-not-merge` tag.

**Pull request merged**
